### PR TITLE
Disentangle the classes of the Audio File Processor

### PR DESCRIPTION
**Pull request by michaelgregorius**
Disentangle the classes of the Audio File Processor by pulling the classes `AudioFileProcessorView` and `AudioFileProcessorWaveView` into their own header and implementation files. This makes the interfaces more concise and easier to reason about the different classes and their purposes.

Remove the friend relationship between `AudioFileProcessor` and `AudioFileProcessorView`. Introduce getters for entities that the view is interested in.

### CODE REVIEW

File: plugins/AudioFileProcessor/AudioFileProcessor.h [hunk 1]
```diff
@@ -77,6 +63,17 @@ class AudioFileProcessor : public Instrument
...
+	Sample const & getSample() const { return m_sample; }
+
+	FloatModel & getAmpModel() { return m_ampModel; }
+	FloatModel & getStartPointModel() { return m_startPointModel; }
+	FloatModel & getEndPointModel() { return m_endPointModel; }
+	FloatModel & getLoopPointModel() { return m_loopPointModel; }
+	BoolModel & getReverseModel() { return m_reverseModel; }
+	IntModel & getLoopModel() { return m_loopModel; }
+	BoolModel & getStutterModel() { return m_stutterModel; }
+	ComboBoxModel & getInterpolationModel() { return m_interpolationModel; }
```
Comment (sakertooth) [inline plugins/AudioFileProcessor/AudioFileProcessor.h:75]: We probably wouldn't use `get` as a prefix here.

**Review by sakertooth — COMMENTED** [review]
Not much to say. One suggestion but not really blocking. You made sure everything still works after the split, correct? I would also format the entire document these are brand new files.

File: plugins/AudioFileProcessor/AudioFileProcessor.h (hunk 1 above)
Comment (michaelgregorius) [reply plugins/AudioFileProcessor/AudioFileProcessor.h]: Adjusted with commit 1aa99e2a802.

**Comment by michaelgregorius** [comment]
> You made sure everything still works after the split, correct?

Yes I made some smoke tests where I loaded a sample, adjusted some controls and played it.

> I would also format the entire document since these are brand new files.

Done with commit adeb975c3e2.

**Review by sakertooth — APPROVED** [review]

**Pull request merged**